slick/src/main/scala/slick/jdbc/DriverDataSource.scala

```diff
@@ -46,42 +46,45 @@ class DriverDataSource(
   // State that gets initialized by `init`
   @volatile private[this] var registered: Boolean = false
   @volatile private[this] var initialized = false
-  private[this] var driver: Driver = _
-  private[this] var connectionProps: Properties = _
+  @volatile private[this] var driver: Driver = _
+  @volatile private[this] var connectionProps: Properties = _
```

Comment:
This should not be needed. `driver` and `connectionProps` are only written within a `synchronized` block. Publishing the writes at the end of that block should be sufficient.